extensions/amp-story-shopping/0.1/amp-story-shopping-tag.js

```diff
@@ -67,6 +68,43 @@ export class AmpStoryShoppingTag extends AMP.BaseElement {
     this.storeService_.subscribe(StateProperty.RTL_STATE, (rtlState) => {
       this.onRtlStateUpdate_(rtlState);
     });
+
+    this.storeService_.subscribe(StateProperty.PAGE_SIZE, () => {
+      this.onPageSizeStateUpdate_();
+    });
+  }
+
+  /**
+   * Helper function to check if the shopping tag should flip if it's too far to the right.
+   * @private
+   */
+  shoudTagFlip_() {
+    const {offsetLeft, offsetWidth} = this.element;
+    this.measureElement(() => {
+      /* We only check the right hand side, as resizing only expands the border to the right. */
+
+      const storyPageWidth =
+        this.element.closest('amp-story-page')./*OK*/ offsetWidth;
```

Comment:
Please get `StateProperty.PAGE_SIZE` from the store service

Reply:
We are not using the width of the Page Size, we are using the width of the story page. The page size will return the width and height of the whole page (including the big border around the mobile rectangle in the middle), so it will not work for desktop if we get StateProperty.PAGE_SIZE from the store service; @processprocess any thoughts? I can't think of a way to calculate that number just using the page size width alone from the store service without involving some sort of DOM measurement.

Reply:
I can confirm that `PAGE_SIZE` is the size of the window and not the page.

[SET_PAGE_SIZE](https://github.com/ampproject/amphtml/blob/main/extensions/amp-story/1.0/amp-story.js#L1497) is currently updated with the [window dimensions](https://github.com/ampproject/amphtml/blob/main/extensions/amp-story/1.0/amp-story.js#L355), which is unexpected in the desktop experience.

I think it would be worthwhile to fix this in a separate PR.
If we were to do that we would also need to update the CSS overrides [here](https://github.com/ampproject/amphtml/blob/main/extensions/amp-story/1.0/amp-story-desktop-one-panel.css#L23-L24).

@gmajoulet Do you think that fix should block this PR? I'm leaning that we should go ahead and fix this now so we can use it in the future.

Reply:
This is 100% a bug, introduced here: https://github.com/ampproject/amphtml/commit/3012bf62aa194478825d34f04b909d6c4cadd061#diff-94f9e2fa2969f98d22c37d9e1f195adbe79ebda93b4ffedf4eab34327238418fL570

Yes, it should be fixed before we move forward with this PR, that should use `PAGE_SIZE`

Reply:
This is bug fixed with #37435